code/modules/projectiles/projectile/bullets.dm

```diff
@@ -183,7 +183,7 @@
 
 /obj/item/projectile/bullet/meteorshot/on_hit(var/atom/target, var/blocked = 0)
 	..()
-	if(istype(target, /atom/movable))
+	if(istype(target, /atom/movable) && !istype(target, /obj/machinery/disposal))

```

Comment:
Instead of this `istype` check (which some would call snowflaked), would a better solution be to call the following: 
- `trunk.remove_trunk_links()` in the `Move` proc for disposals units; 
- `remove_trunk_links()` in the `Move` proc for trunks?

This should fix all edge cases of disposals chutes moving.

Reply:
Would also fix it yeah. I don't know though if it's preferable that you can move attached disposal units. Another fix might be to break (deconstruct) them if they move. Need some feedback on this one. Thanks for the tips!

Reply:
Having stuff in the `Move` proc doesn't mean the thing needs to move (or imply that they move in the first place) -  this would be handled on the `anchored` var anyway. But all atoms have a `Move` proc which means you can change what happens if it does move. In this case it would break the trunk connection

You could also just destroy it if it moves, like you say. But I'd just break the connection